lib/modules/dosomething/dosomething_payment/dosomething_payment.module

```diff
@@ -214,7 +227,7 @@ function dosomething_payment_form_submit($form, &$form_state) {
   // Expected amount to charge to Stripe should be in cents.
   $values['amount'] = $values['amount'] * 100;
   // Send Payment request to Stripe API.
-  if (dosomething_payment_post_payment_stripe($values)) {
+  if (dosomething_payment_post_payment_stripe($values) && !isset($values['email'])) {
```

Comment:
Has this been tested? I think this will still send the payment information to Stripe even if the `$values['email']` is set. If you move the `isset()` check first, it should short-circuit.